profiler/src/ProfilerEngine/Datadog.Profiler.Native/GroupSampler.h

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#pragma once
+
+#include <unordered_set>
+
+#include "GenericSampler.h"
+#include "IConfiguration.h"
+
+
+// Template class that support "sampling by group".
+// At least one element in the group will ALWAYS be sampled
+template <class TGroup>
+class GroupSampler : public GenericSampler
+{
+public:
+    GroupSampler<TGroup>(int32_t samplesLimit, int32_t uploadIntervalMs)
+        :
+        GenericSampler(samplesLimit, uploadIntervalMs)
+    {
+    }
+
+    bool Sample(TGroup group)
+    {
+        std::unique_lock lock(_knownGroupsMutex);
+
+        if (_knownGroups.find(group) == _knownGroups.end())
+        {
+            // This is the first time we see this group in this time window,
+            // force the sampling decision
+            _knownGroups.insert(group);
+
+            return _sampler.Keep();
+        }
```

Comment:
you can use only one call to `insert` to avoid two requests the set.
```
auto [it, inserted] = _knownGroups.insert(group);
if (inserted)
{
   return _sampler.Keep();
}
...
```

Reply:
✅ done